### PR TITLE
Feature/hamburger menu

### PR DESCRIPTION
**Pull request by ninecodes**
- Added Basic Hamburger Menu regarding issue #214 
- Show on smaller screen sizes on the right side of the screen
image

- Basic UI styling for menu items:
image

### CODE REVIEW

**Comment by goncy** [comment]
Hey @ninecodes, this looks awesome! Thanks for it, the only thing we would like (not fully related to this feature), is to make the sidebar full width on mobile, do you want to do it? Otherwise we can merge it and I can do it later

**Comment by ninecodes** [comment]
> Hey @ninecodes, this looks awesome! Thanks for it, the only thing we would like (not fully related to this feature), is to make the sidebar full width on mobile, do you want to do it? Otherwise we can merge it and I can do it later

Thanks for the feedback, just pushed the mobile full width styling :) hope this is fine :)

**Comment by goncy** [comment]
@ninecodes It looks great, thanks! Can you fix the conflicts? (they are just prettier conflicts) so we can get this merged now

File: components/common/UserNav/MenuSidebarView/MenuSidebarView.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+interface MenuProps {
+  links?: Link[]
+}
+
+const MenuSidebarView: FC<MenuProps> = (props) => {
+  const { closeSidebar } = useUI()
+  const handleClose = () => closeSidebar()
+
+  console.log(props.links)
```
Comment (goncy) [inline components/common/UserNav/MenuSidebarView/MenuSidebarView.tsx:20]: Can you remove this log?

File: components/common/Layout/Layout.tsx [hunk 2]
```diff
@@ -69,12 +70,14 @@ const ModalUI: FC = () => {
...
 }
 
-const SidebarView: FC<{ sidebarView: string; closeSidebar(): any }> = ({
-  sidebarView,
-  closeSidebar,
-}) => {
+const SidebarView: FC<{
+  sidebarView: string
+  closeSidebar(): any
+  links: any
```
Comment (goncy) [inline components/common/Layout/Layout.tsx:81]: If its possible we can create a `Link` interface just on top, something like:

```ts
interface Link{
  href: string
  label: string
}
```

And do `links: Link[]`

**Comment by ninecodes** [comment]
> @ninecodes It looks great, thanks! Can you fix the conflicts? (they are just prettier conflicts) so we can get this merged now

Just resolved them, hope its looking good now :) @goncy

File: components/common/Layout/Layout.tsx [hunk 3]
```diff
@@ -83,10 +86,14 @@ const SidebarView: FC<{ sidebarView: string; closeSidebar(): any }> = ({
   )
 }
 
-const SidebarUI: FC = () => {
+const SidebarUI: FC<{ links: any }> = ({ links }) => {
```
Comment (goncy) [inline components/common/Layout/Layout.tsx:94]: You can even extract the links to its own type or interface (or import it from other place) and use that type in here too

**Review by goncy — CHANGES_REQUESTED** [review]
This are some small changes if possible!

**Comment by goncy** [comment]
Sorry about the back and forth @ninecodes, really appreciate your help 💪

**Comment by ninecodes** [comment]
> Sorry about the back and forth @ninecodes, really appreciate your help 💪

No worry @goncy just implemented your suggested changes 👍🏼 Hope this helps the project move forward

**Review by goncy — APPROVED** [review]

**Comment by goncy** [comment]
Awesome @ninecodes thanks! Merged 🙌

**Comment by ninecodes** [comment]
> Awesome @ninecodes thanks! Merged 🙌

Nice to hear that, will be contributing to other features as well from now on 👍🏼 @goncy

**Comment by ViktorYermak** [comment]
This mobile menu, like mobile menus in general, is part of an outdated paradigm. A modern user interface should be as visually simple as possible; no hamburgers, no drawers. Therefore, the mobile menu should be optional!

**Pull request merged**